gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/DefaultHttpAcceptor.java

```diff
@@ -101,12 +116,16 @@ public int priority() {
 
     @Override
     public boolean accept(Request request) {
-        return accept(request.host(), request.path());
+        return accept(request.host(), request.path(), null);
     }
 
     @Override
-    public boolean accept(String host, String path) {
-        return matchHost(host) && matchPath(path);
+    public boolean accept(String host, String path, String serverId) {
+        return matchServer(serverId) && matchHost(host) && matchPath(path);
+    }
+
+    private boolean matchServer(String serverId) {
+        return serverIds == null || serverId == null || serverIds.contains(serverId);
```

Comment:
if serverIds is not null but the given serverId is null, shouldn't we reject the demand? Something like bellow 
```suggestion
        return serverIds == null || (serverId != null && serverIds.contains(serverId));
```

Reply:
Yes, sounds definitely better